internal/controller/services/monitoring/monitoring_controller_actions.go

```
@@ -135,3 +137,53 @@ func checkDSCI(ctx context.Context, rr *odhtypes.ReconciliationRequest) (bool, *
 
 	return true, dsci, nil
 }
+
+// deployTempo creates Tempo resources based on the Monitoring CR configuration.
+func deployTempo(ctx context.Context, rr *odhtypes.ReconciliationRequest) error {
+	monitoring, ok := rr.Instance.(*serviceApi.Monitoring)
+	if !ok {
+		return errors.New("instance is not of type *services.Monitoring")
+	}
+
+	// Read traces configuration directly from Monitoring CR
+	if monitoring.Spec.Traces == nil {
+		// No traces configuration - GC action will clean up any existing Tempo resources
+		return nil
+	}
+
+	traces := monitoring.Spec.Traces
+
+	var requiredCRD schema.GroupVersionKind
+	if traces.Storage.Backend == "pv" {
+		requiredCRD = gvk.TempoMonolithic
+	} else {
+		requiredCRD = gvk.TempoStack
+	}
+
+	crdExists, _ := cluster.HasCRD(ctx, rr.Client, requiredCRD)
```

I would avoid to swallow the error here